db/external_sst_file_ingestion_job.cc

```diff
@@ -370,7 +370,6 @@ Status ExternalSstFileIngestionJob::NeedsFlush(bool* flush_needed,
 Status ExternalSstFileIngestionJob::Run() {
   Status status;
   SuperVersion* super_version = cfd_->GetSuperVersion();
-#ifndef NDEBUG
```

Comment:
I'm also making this change here to not just run this piece of code in debug mode. This race condition showed that previously, the job could be using outdated SuperVersion for level assignment, which can be very harmful.

Reply:
I'm not sure about this since this is done while blocking all the writes and holding mutex. With your fix it should not happen, and our stress test has proven to be able to catch it if it happens?

Reply:
Since this is less likely to be an issue after the fix. I reverted this change.

Reply:
We can do cheaper checks like checking that this SV should not have any immutable memtable if flush was called.

Reply:
Thanks for this idea! I took a look at this and there is some subtle handling regarding dropped column families, especially for atomic flush. Let me look into this more and do it in a follow up.